src/ck-ldb/CentralLB.C

```diff
@@ -419,35 +419,24 @@ void CentralLB::MissMigrate(int waitForBarrier)
 void CentralLB::buildStats()
 {
     statsData->nprocs() = stats_msg_count;
-    // allocate space
-    statsData->objData.resize(statsData->n_objs);
-    statsData->from_proc.resize(statsData->n_objs);
-    statsData->to_proc.resize(statsData->n_objs);
-    statsData->commData.resize(statsData->n_comm);
-
-    int nobj = 0;
-    int ncom = 0;
-    int nmigobj = 0;
-    // copy all data in individule message to this big structure
+
+    // copy all data in individual messages to this big structure
+    // Space has already been reserved in ReceiveStats
     for (int pe=0; pe<CkNumPes(); pe++) {
        int i;
        CLBStatsMsg *msg = statsMsgsList[pe];
        if(msg == NULL) continue;
-       for (i=0; i<msg->n_objs; i++) {
-         statsData->from_proc[nobj] = statsData->to_proc[nobj] = pe;
-	 statsData->objData[nobj] = msg->objData[i];
-         if (msg->objData[i].migratable) nmigobj++;
-	 nobj++;
-       }
-       for (i=0; i<msg->n_comm; i++) {
-	 statsData->commData[ncom] = msg->commData[i];
-	 ncom++;
-       }
+       statsData->objData.insert(statsData->objData.end(), msg->objData, msg->objData + msg->n_objs);
+       statsData->from_proc.insert(statsData->from_proc.end(), msg->n_objs, pe);
+       statsData->to_proc.insert(statsData->to_proc.end(), msg->n_objs, pe);
+       statsData->commData.insert(statsData->commData.end(), msg->commData, msg->commData + msg->n_comm);
        // free the memory
        delete msg;
        statsMsgsList[pe]=0;
     }
-    statsData->n_migrateobjs = nmigobj;
+    statsData->n_migrateobjs = count_if(statsData->objData.begin(),
```

Comment:
Maybe more consistent to do "std::count_if" here or to do "count_if" below

Reply:
Agreed, added `std::` to all uses.